resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala

```diff
@@ -198,4 +198,13 @@ class KubernetesConfSuite extends SparkFunSuite {
     assert(driverConf.nodeSelector === CUSTOM_NODE_SELECTOR)
     assert(driverConf.driverNodeSelector === CUSTOM_DRIVER_NODE_SELECTOR)
   }
+
+  test("resourceName prefix") {
+    val name = "a" + "b" * 100 + "c"
+    assert(KubernetesConf.getResourceNamePrefix(name).startsWith(name + "-"))
+
+    val expected = "a" + "b" * (KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH - 1 - 1 - 16)
+    assert(KubernetesConf.getResourceNamePrefix(name, KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH)
+      .startsWith(expected + "-"))
```

Comment:
What about extending this test with the cases which covers the bounds like when `getResourceNamePrefix` is called with a string which length
- is below `KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH` with one char
- is the same as `KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH`
- it is longer with one char